In [3]:
# 创建一个简单的DataFrame数据
import pandas as pd 

frame = pd.DataFrame([[1,2,3,3],[4,5,6,7],[9,8,7,3]],index=['a','b','c'],columns=['ball','pen','pencil','paper'])
frame

ball  pen  pencil  paper
a     1    2       3      3
b     4    5       6      7
c     9    8       7      3

In [19]:
# 写入DataFrame创建好的DateFrame数据到myW_01.csv文件中在查看
# 写入文件用to_csv("")，里面的参数代表文件名

frame.to_csv("myW_01.csv",index=False,header=False) # 可以取消写入索引很列名称

readCSV = pd.read_csv('myW_01.csv',header=None)
readCSV


0  1  2  3
0  1  2  3  3
1  4  5  6  7
2  9  8  7  3

In [4]:
import numpy as np 

frame1 = pd.DataFrame([[1,2,np.NaN,3],[4,np.NaN,np.NaN,7],[9,8,7,np.NaN]],
                      index=['a','b','c'],columns=['ball','pen','pencil','paper'])
frame1.to_csv('myW_02.csv')   # NaN写入文件后，显示为空字段
frame1

ball  pen  pencil  paper
a     1  2.0     NaN    3.0
b     4  NaN     NaN    7.0
c     9  8.0     7.0    NaN

In [6]:
# 在to_csv()函数中使用na_rep选项把空字段替换为需要的值，常用值由NULL、0和NaN
frame1.to_csv('myW_03.csv',na_rep="NaN")

In [8]:
import pandas as pd
   
# 三个字段 name, site, age
nme = ["Google", "Runoob", "Taobao", "Wiki"]
st = ["www.google.com", "www.runoob.com", "www.taobao.com", "www.wikipedia.org"]
ag = [90, 40, 80, 98]
   
# 字典
dict = {'name': nme, 'site': st, 'age': ag}
     
df = pd.DataFrame(dict)
 
# 保存 dataframe
df.to_csv('myW_site.csv')

In [9]:
pd.read_csv("myW_site.csv")


Unnamed: 0    name               site  age
0           0  Google     www.google.com   90
1           1  Runoob     www.runoob.com   40
2           2  Taobao     www.taobao.com   80
3           3    Wiki  www.wikipedia.org   98

In [15]:
# 数据处理
# head( n ) 方法用于读取前面的 n 行，如果不填参数 n ，默认返回 5 行
import pandas as pd 

df = pd.read_csv('nba.csv')
print(df.head())


            Name            Team  Number Position   Age Height  Weight  \
0  Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
1    Jae Crowder  Boston Celtics    99.0       SF  25.0    6-6   235.0   
2   John Holland  Boston Celtics    30.0       SG  27.0    6-5   205.0   
3    R.J. Hunter  Boston Celtics    28.0       SG  22.0    6-5   185.0   
4  Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   

             College     Salary  
0              Texas  7730337.0  
1          Marquette  6796117.0  
2  Boston University        NaN  
3      Georgia State  1148640.0  
4                NaN  5000000.0  


In [17]:
# tail()
# tail( n ) 方法用于读取尾部的 n 行，
# 如果不填参数 n ，默认返回 5 行，空行各个字段的值返回 NaN。
print(df.tail(5))

             Name       Team  Number Position   Age Height  Weight College  \
453  Shelvin Mack  Utah Jazz     8.0       PG  26.0    6-3   203.0  Butler   
454     Raul Neto  Utah Jazz    25.0       PG  24.0    6-1   179.0     NaN   
455  Tibor Pleiss  Utah Jazz    21.0        C  26.0    7-3   256.0     NaN   
456   Jeff Withey  Utah Jazz    24.0        C  26.0    7-0   231.0  Kansas   
457           NaN        NaN     NaN      NaN   NaN    NaN     NaN     NaN   

        Salary  
453  2433333.0  
454   900000.0  
455  2900000.0  
456   947276.0  
457        NaN  


In [19]:
# info()
# info() 方法返回表格的一些基本信息：
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 458 entries, 0 to 457
Data columns (total 9 columns):
Name        457 non-null object
Team        457 non-null object
Number      457 non-null float64
Position    457 non-null object
Age         457 non-null float64
Height      457 non-null object
Weight      457 non-null float64
College     373 non-null object
Salary      446 non-null float64
dtypes: float64(4), object(5)
memory usage: 32.3+ KB
None



RangeIndex: 458 entries, 0 to 457          # 行数，458 行，第一行编号为 0
Data columns (total 9 columns):            # 列数，9列
   Column    Non-Null Count  Dtype       # 各列的数据类型
---  ------    --------------  -----  
 0   Name      457 non-null    object 
 1   Team      457 non-null    object 
 2   Number    457 non-null    float64
 3   Position  457 non-null    object 
 4   Age       457 non-null    float64
 5   Height    457 non-null    object 
 6   Weight    457 non-null    float64
 7   College   373 non-null    object    # non-null，意思为非空的数据    
 8   Salary    446 non-null    float64
dtypes: float64(4), object(5)                 # 类型
